playbooks/roles/dev-patcher/tasks/main.yml

```diff
@@ -29,6 +29,17 @@
 
 - name: Get patch details
   include_tasks: patch-repos.yml
-  loop: "{{ dev_patcher_patches }}"
+  loop: "{{ dev_patcher_patches | default([], True) }}"
   loop_control:
     loop_var: patchnumber
+
+- name: Check if there is local code change to cloned repos on deployer
+  stat:
+    path: "{{ role_path }}/files/repos/"
+  register: local_change_repos
+
+- name: Copy local code changes to cloned repos on deployer
+  synchronize:
```

Comment:
I am not sure why this exists? The feature currently works, no need for sync....

Reply:
This is quite a life-saver when we are developing/ triaging issue locally. 

So I can make local changes in specific files of airship or osh project and copy those files and file structure to files/repos/ path. Then those changes are used and made available in my actual deployment. I heavily use it and 
can see benefit for others as well.

Reply:
So you want a new feature in developer mode to override local changes without an upstream patch. I don't think this should be done in this commit. I could be fine in another PR.

Reply:
We already have this change in ucp branch and used. Having a separate PR for every UCP branch change is going to inefficient, error-prone and time-consuming. I don't see any downside of having this change slide-in or added benefit of having as separate PR as its already reviewed/used multiple times.

Reply:
Ok for me, if the others are okay. I will not be pedantic here.

Reply:
kind of ok for me, can you add tags to the tasks so we can skip them if needed?

Would have loved to have this as a different Pr as this is a different logical change, shouldnt be paired here as its a new fature.

Reply:
There is check if file/repos/ is not there, then nothing is executed .

Not sure about which tag you are referring to use here. 

Also making changes here trigger change in ucp branch as it needs to be re-merged after this PR is merged. Trying to minimize merge-conflict in a way